src/modules/esl-utils/decorators/prop.ts

```diff
@@ -1,11 +1,32 @@
+import type {PropertyProvider} from '../misc/functions';
+
 /** Property configuration */
 export type OverrideDecoratorConfig = {
-  /** To define enumerable property */
-  enumerable?: boolean;
   /** To define readonly property */
   readonly?: boolean;
+  /** To define enumerable property */
+  enumerable?: boolean;
 };
 
+/** Builds getter from provider */
+function getter<T>(provider: PropertyProvider<T>) {
+  return function (this: any): T {
+    return provider.call(this, this);
+  };
+}
+
+/** Builds own property setter */
+function setter(name: string, readonly?: boolean) {
+  if (readonly) return (): void => void 0;
+  return function (value: any): void {
+    Object.defineProperty(this, name, {
+      value,
+      writable: true,
+      configurable: true
+    });
```

Comment:
Should it be enumerable?

Reply:
That's a good (and tricky) question, by the way... 

We have an enumerable option for the original property, which is not a used characteristic that was broken in addition.
Note that even if we pass enumerable: true, we will not get it enumerable on the instance; we just make it such on the prototype.
So, I'm not sure we chose the right strategy to support it at all.

For instance, the override that we are watching for, I believe it's an even more realistic case to make it enumerable, but I don't think it's popular.

So let's create an issue to come back to that question...